proto/aep-api/aep/api/field_behavior.proto

```diff
@@ -0,0 +1,90 @@
+syntax = "proto3";
+
+package aep.api;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "aep.dev/api";
+option java_multiple_files = true;
+option java_outer_classname = "FieldBehaviorProto";
+option java_package = "com.aep.api";
+option objc_class_prefix = "AEP";
+
+extend google.protobuf.FieldOptions {
+  // A designation of a specific field behavior (required, output only, etc.)
+  // in protobuf messages.
+  //
+  // Examples:
+  //
+  //   string name = 1 [(aep.api.field_behavior) = REQUIRED];
+  //   State state = 1 [(aep.api.field_behavior) = OUTPUT_ONLY];
+  //   aep.protobuf.Duration ttl = 1
+  //     [(aep.api.field_behavior) = INPUT_ONLY];
+  //   aep.protobuf.Timestamp expire_time = 1
+  //     [(aep.api.field_behavior) = OUTPUT_ONLY,
+  //      (aep.api.field_behavior) = IMMUTABLE];
+  repeated aep.api.FieldBehavior field_behavior = 1052 [packed = false];
+}
+
+// An indicator of the behavior of a given field (for example, that a field
+// is required in requests, or given as output but ignored as input).
+// This **does not** change the behavior in protocol buffers itself; it only
+// denotes the behavior and may affect how API tooling handles the field.
+//
+// Note: This enum **may** receive new values in the future.
+enum FieldBehavior {
+  // Conventional default for enums. Do not use this.
+  FIELD_BEHAVIOR_UNSPECIFIED = 0;
+
+  // Specifically denotes a field as optional.
+  // While all fields in protocol buffers are optional, this may be specified
+  // for emphasis if appropriate.
+  OPTIONAL = 1;
```

Comment:
kind of hilarious that the aep protos themselves are not proto compliant. I kind of find FIELD_BEHAVIOR_OPTIONAL ugly myself, but I guess we don't want conflicts in generated libraries.